gluonnlp/data/language_model.py

```diff
@@ -251,3 +254,89 @@ def __init__(self, segment='train', skip_empty=True,
         super(WikiText103Raw,
               self).__init__('wikitext-103', segment, bos, eos, skip_empty,
                              root, tokenizer=tokenizer, **kwargs)
+
+class _GBWStream(LanguageModelStream):
+    def __init__(self, namespace, segment, bos, eos, skip_empty, root):
+        """Directory layout:
+           - root ($MXNET_HOME/datasets/gbw)
+             - archive_file (1-billion-word-language-modeling-benchmark-r13output.tar.gz)
+             - dir (1-billion-word-language-modeling-benchmark-r13output)
+               - subdir (training-monolingual.tokenized.shuffled)
+               - subdir (heldout-monolingual.tokenized.shuffled)
+        """
+        root = os.path.expanduser(root)
+        if not os.path.isdir(root):
+            os.makedirs(root)
+        self._root = root
+        self._dir = os.path.join(root, '1-billion-word-language-modeling-benchmark-r13output')
+        self._namespace = 'gluon/dataset/{}'.format(namespace)
+        subdir_name, pattern, data_hash = self._data_file[segment]
+        self._subdir = os.path.join(self._dir, subdir_name)
+        self._file_pattern = os.path.join(self._subdir, pattern)
+        self._data_hash = data_hash
+        self._get_data()
+        sampler = 'random' if segment == 'train' else 'sequential'
```

Comment:
why sequential is needed? In the evaluation, the model should also use random sampler.

Reply:
i'll email author to figure it out..

Reply:
Thanks, the first author and second author have different ways of evaluating the model...